### PR TITLE
docs: Update sync_dist warning for multiple processes

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

`sync_fn` is not a parameter for `self.log`.

Related to #6716 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

**Review by carmocca — COMMENTED** [review]
Could instead mention `sync_dist_op`

https://github.com/PyTorchLightning/pytorch-lightning/blob/7f91c5ebbc60b5ebed61072846da606f5d3957e6/pytorch_lightning/core/lightning.py#L222

**Review by s-rog — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py [hunk 1]
```diff
@@ -133,8 +133,8 @@ def run_epoch_func(self, results, opt_metric, func_name, *args, **kwargs) -> Non
                         warning_cache.warn(
                             f"The value associated to the key {non_metric_key}: {metric.cpu().tolist()} "
                             "doesn't appear to be the same accross all processes. "
-                            "HINT: One could either do: `self.log(..., sync_dist=True, sync_fn=torch.mean)`"
-                            " to force mean reduction across processes which can be inaccurate or implement"
+                            "HINT: One could either do: `self.log(..., sync_dist=True)` to force mean"
+                            " reduction by default across processes which can be inaccurate or implement"
                             " a `torchmetrics.Metric`"
```
Comment (tchaton) [inline pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py:138]:
```suggestion
                            " a rely on `torchmetrics.Metric`"
```

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py:138]: "implement a rely on `torchmetrics.Metric`"?

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
